frappe/desk/page/user_profile/user_profile.js

```diff
@@ -0,0 +1,472 @@
+frappe.provide('frappe.energy_points');
+
+frappe.pages['user-profile'].on_page_load = function(wrapper) {
+
+	frappe.ui.make_app_page({
+		parent: wrapper,
+		title: __('User Profile'),
+	});
+
+	let user_profile = new UserProfile(wrapper);
+	$(wrapper).bind('show', ()=> {
+		user_profile.show();
+	});
+};
+
+class UserProfile {
+
+	constructor(wrapper) {
+		this.wrapper = $(wrapper);
+		this.page = wrapper.page;
+		this.sidebar = this.wrapper.find('.layout-side-section');
+		this.main_section = this.wrapper.find('.layout-main-section');
+	}
+
+	show() {
+		this.route = frappe.get_route();
+		//validate if user
```

Comment:
Set user first, then validate, if true make profile
`this.user_id = frappe.get_route[1] || frappe.session.user`